docs/static/persistent-queues.asciidoc

```diff
@@ -101,23 +103,24 @@ queue.max_bytes: 4gb
 [[backpressure-persistent-queues]]
 ==== Handling Back Pressure
 
-Logstash has a built-in mechanism that exerts back pressure on the data flow 
-when the queue is full. This mechanism helps Logstash control the rate of data
-flow at the input stage without overwhelming downstream stages and outputs like
+When the queue is full, Logstash puts back pressure on the inputs to stall data
+flowing into Logstash. This mechanism helps Logstash control the rate of data
+flow at the input stage without overwhelming outputs and outputs like
 Elasticsearch.
 
-You can control when back pressure happens by using the `queue.max_bytes` 
-setting to configure the capacity of the queue on disk. The following example
-sets the total capacity of the queue to 8gb:
+Use `queue.max_bytes` setting to configure the total capacity of the queue on
+disk. The following example sets the total capacity of the queue to 8gb:
 
 [source, yaml]
 queue.type: persisted
 queue.max_bytes: 8gb
 
-With these settings specified, Logstash will buffer unACKed events on disk until 
-the size of the queue reaches 8gb. When the queue is full of unACKed events, and
+With these settings specified, Logstash will buffer events on disk until the
+size of the queue reaches 8gb. When the queue is full of unACKed events, and
 the size limit has been reached, Logstash will no longer accept new events. 
 
+On disk, the queue is stored as a set of pages where each page is one file. Each page can be at most `queue.page_capacity` in size. Pages are deleted (garbage collected) once all events in that page have been ACKed. If an older page has at least one event that is not yet ACKed, that entire page will remain on disk until all events in that page are successfully processed. Each page containing unprocessed events will count against the `queue.max_bytes` byte size.
```

Comment:
the situation where a page is not deleted because all events are not yet acked can happen in these 2 scenarios:
1- outputs are slower than inputs, which will typically result in the accumulation of multiple pages and all events in the page(s) will be unacked until the outputs are ready to process a page. Typically there will be a short overlap when some output start processing a new page and other output finish processing the previous page.
2- an output (or multiple outputs) are "stuck" in a retry loop for some time and some events will remain unacked during this retry period so a page might be kept "alive" (not deleted) because of this particular output "stuck" in retry for some time while other output might actually finish processing the rest of the page and move on to the other page. This situation will be temporary up to be output retry thresholds which is plugin dependant.